tests/test_db.py

```diff
@@ -97,3 +106,318 @@ def test_register_db(tmp_path):
+
+    db_path = tmp_path / "prova.json"
+    db.register(db_path, url_zst, 1)
+
+    responses.add(
+        responses.HEAD,
+        url_zst,
+        status=404,
+        headers={"ETag": "123", "Accept-Encoding": "zstd"},
+    )
+
+    responses.add(
+        responses.GET,
+        url_zst,
+        status=404,
+        body=requests.exceptions.HTTPError("HTTP error"),
+    )
+
+    responses.add(
+        responses.HEAD,
+        url_xz,
+        status=200,
+        headers={"ETag": "123", "Accept-Encoding": "xz"},
+    )
+
+    mock_xz(db_path.with_suffix(db_path.suffix + ".xz"))
+
+    with open(db_path.with_suffix(db_path.suffix + ".xz"), "rb") as content:
+        responses.add(responses.GET, url_xz, status=200, body=content.read())
+
+    os.remove(db_path.with_suffix(db_path.suffix + ".xz"))
+
+    db.download(db_path)
+
+    assert os.path.exists(db_path)
+    assert os.path.exists(db_path.with_suffix(db_path.suffix + ".xz"))
+    assert os.path.exists(db_path.with_suffix(db_path.suffix + ".xz.etag"))
+
+
+@responses.activate
+def test_download_bad_format(tmp_path):
+    url_zst = "https://index.taskcluster.net/v1/task/project.relman.bugbug.data_commits.latest/artifacts/public/commits.json.zst"
+    url_xz = "https://index.taskcluster.net/v1/task/project.relman.bugbug.data_commits.latest/artifacts/public/commits.json.xz"
+
+    db_path = tmp_path / "prova.json"
+    db.register(db_path, url_zst, 1)
+
+    responses.add(
+        responses.HEAD,
+        url_zst,
+        status=404,
+        headers={"ETag": "123", "Accept-Encoding": "zstd"},
+    )
+
+    responses.add(
+        responses.GET,
+        url_zst,
+        status=404,
+        body=requests.exceptions.HTTPError("HTTP error"),
+    )
+
+    responses.add(
+        responses.HEAD,
+        url_xz,
+        status=404,
+        headers={"ETag": "123", "Accept-Encoding": "xz"},
+    )
+
+    responses.add(
+        responses.GET,
+        url_xz,
+        status=404,
+        body=requests.exceptions.HTTPError("HTTP error"),
+    )
+
+    with pytest.raises(requests.exceptions.HTTPError):
+        db.download(db_path)
+
+
+@responses.activate
+def test_download_support_file_zst(tmp_path, mock_zst):
+    url_zst = "https://index.taskcluster.net/v1/task/project.relman.bugbug.data_commits.latest/artifacts/public/commits.json.zst"
+    support_filename = "support.zst"
+    url = urljoin(url_zst, support_filename)
+
+    db_path = tmp_path / "prova.json"
+    db.register(db_path, url_zst, 1, support_files=[support_filename])
+
+    responses.add(
+        responses.HEAD,
+        url,
+        status=200,
+        headers={"ETag": "123", "Accept-Encoding": "zstd"},
+    )
+
+    mock_zst(db_path.with_suffix(db_path.suffix + ".zst"))
+
+    with open(db_path.with_suffix(db_path.suffix + ".zst"), "rb") as content:
+        responses.add(responses.GET, url, status=200, body=content.read())
+
+    os.remove(db_path.with_suffix(db_path.suffix + ".zst"))
```

Comment:
Why are you manually removing the DB here?

Reply:
> Why are you manually removing the DB here?

I created that mock .zst file just for the body of the mock request, and removed it because the next step is to test the ability to download the compressed file.

Reply:
You can create the mock file in another path, so you don't have to remove it.

Reply:
Just updated the tests to save those mock files to another temporary path, and changed the test names to __missing.